src/test/kotlin/com/atlassian/performance/tools/awsinfrastructure/SuppressingRunnableTest.kt

```diff
@@ -0,0 +1,69 @@
+package com.atlassian.performance.tools.awsinfrastructure
+
+import org.hamcrest.CoreMatchers.*
+import org.junit.Assert.*
+import org.junit.Test
+
+class SuppressingRunnableTest {
+    @Test
+    fun shouldExecuteAllEvenIfFirstFails() {
+        var executed1 = false
+        var executed2 = false
+        var executed3 = false
+        val runnable = SuppressingRunnable(
+            listOf(
+                Runnable { executed1 = true; throw Exception("Fail 1") },
+                Runnable { executed2 = true; throw Exception("Fail 2") },
+                Runnable { executed3 = true; throw Exception("Fail 3") }
+            )
+        )
+
+        try {
```

Comment:
Nit (would require bumping JUnit to v4.13):
`assertThrows(Exception::class.java) { runnable.run() }
`

Reply:
Not sure if I want to assert on the exception here. I believe such assert would be better to be in another test. That's esspecially because adding another unit test has negligible execution cost and is able to separate requirements.

Reply:
If we want to reduce this line to single one we could also use Kotlin's [runCatching](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/run-catching.html), however it would also require dependency bump, as it's available from Kotlin 1.3 and we have 1.2 (IIRC there is some compatibility reason why we are not up to date).

I'm not sure if it's worth the effort.